src/lib.rs

```diff
@@ -131,13 +131,9 @@ impl<T: AsMut<[u8]>> LogBuffer<T> {
             byte & 0b11111000 == 0b11110000
         }
 
-        let buffer = self.buffer.as_mut();
-        for i in 0..buffer.len() {
-            if is_utf8_leader(buffer[i]) {
-                return core::str::from_utf8(&buffer[i..]).unwrap()
-            }
-        }
-        return ""
+        let buffer = &*self.buffer.as_mut();
+        buffer.iter().cloned().position(is_utf8_leader).map_or("", |i|
```

Comment:
The `&*` is an immutable reborrow because we need to use buffer twice, first for iterating and second for slicing. The `.cloned()` is the canonical way to transform our `Iterator<Item=&u8>` to an `Iterator<Item=u8>`.